src/accommodation/accommodation.service.ts

```diff
@@ -381,4 +410,17 @@ export class AccommodationService {
 
     return countByRating;
   }
+
+  private generateOderingArray<T>(array: [value: SortOrder | undefined, columnName: keyof T][]) {
```

Comment:
looks like value: SortOrder? is enough for this case

Reply:
no it is not enough here. value can be undefined, if the front does not send it, and inside this func i check, If a value is truthy, i will apply the ordering, but if the value is falsy i do not need to apply it 

For example:
If front sends req like this 
```http://localhost:3000/api/accommodations/00324ac1-b0f8-48c7-926c-ba97a9a2d213/reviews?orderByRate=asc&limit=12&page=1
```
i do not have the orderByDate field here, and in my func checks, if it is undefined it do not apply this.
If i remove undefined and check for this my prisma query will look like this:
```js
orderBy: [
        {
          createdAt: undefined,
        },
        {
          rate: 'asc',
        },
      ];
```
And it does not apply to Prisma 

Or do you have any suggestions for this??

Reply:
yep, there is difference between key? and  key | undefined. If it necessary then leave as is